tomviz/DataSource.cxx

```diff
@@ -56,6 +56,31 @@
 #include <cstring>
 #include <sstream>
 
+namespace {
+void createOrResizeTiltAnglesArray(vtkDataObject* data)
+{
+  auto fd = data->GetFieldData();
+  if (!fd->HasArray("tilt_angles")) {
+    int* extent = vtkImageData::SafeDownCast(data)->GetExtent();
+    int numTiltAngles = extent[5] - extent[4] + 1;
+    vtkNew<vtkDoubleArray> array;
+    array->SetName("tilt_angles");
+    array->SetNumberOfTuples(numTiltAngles);
+    array->FillComponent(0, 0.0);
+    fd->AddArray(array);
+  } else {
+    // if it exists, ensure the size of the tilt angles array
+    // corresponds to the size of the data
+    int* extent = vtkImageData::SafeDownCast(data)->GetExtent();
+    int numTiltAngles = extent[5] - extent[4] + 1;
+    auto array = fd->GetArray("tilt_angles");
+    if (numTiltAngles != array->GetNumberOfTuples()) {
+      array->SetNumberOfTuples(numTiltAngles);
+    }
+  }
+}
+} // namespace
+
```

Comment:
Whats the reason for moving this outside the internals object?

Reply:
I need to call this from a static member function where I don't have an internals object (`setTiltAngles`)